3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -64,36 +66,26 @@ function renderError(error) {
 
 async function main() {
   try {
-    const data = await fetchData(`https://pokeapi.co/api/v2/pokemon?limit=151`);
-
     const buttonGetPokemons = document.createElement('button');
     const selectPokemons = document.createElement('select');
-    const imagePokemon = document.createElement('img');
 
-    buttonGetPokemons.id = 'btn';
+    buttonGetPokemons.classList.add('btn');
+    buttonGetPokemons.setAttribute('type', 'submit');
     buttonGetPokemons.textContent = 'Get Pokemons';
     selectPokemons.id = 'select-Pokemons';
-    imagePokemon.id = 'pokemon-image';
 
     document.body.appendChild(buttonGetPokemons);
     document.body.appendChild(selectPokemons);
-    document.body.appendChild(imagePokemon);
 
-    buttonGetPokemons.addEventListener('click', () => {
-      data.results.forEach((pokemon) => {
-        fetchAndPopulatePokemons(pokemon.name);
-      });
-
-      selectPokemons.addEventListener('change', (e) => {
-        data.results.forEach((pokemon) => {
-          if (pokemon.name === e.target.value) {
-            fetchImage(pokemon.url);
-          }
-        });
-      });
+    buttonGetPokemons.addEventListener('click', fetchAndPopulatePokemons());
```

Comment:
The function `fetchAndPopulatePokemons()` is called here immediately (because you added parentheses).  The function returns `undefined` which is what `.addEventListener` gets as it parameter, which effectively does nothing.

```suggestion
    buttonGetPokemons.addEventListener('click', fetchAndPopulatePokemons);
```

If you fix this error, make sure also that if the button is pressed more than once that the fetched pokemons replace the existing list rather than adding to it.